Adds phpunit, phpcs and sets up GH actions to run tests

jazzsequence opened the pull request:
Does what it says on the tin. Sets up PHPCS (via `drupal/coder`) and PHPUnit and adds GH actions workflow to automate tests.

Fixes linting issues that came up after installing the Drupal Coder package and running phpcs.

This PR also updates the README file with the Composer commands for linting, `phpcbf` and testing.

Fixes #3

jazzsequence committed:
add gitignore

jazzsequence committed:
require phpunit

jazzsequence committed:
add phpunit and drupal/coder (phpcs)

jazzsequence committed:
add phpunit xml file

jazzsequence committed:
require the composer autoloader for tests

jazzsequence committed:
add simple, method_exists tests

jazzsequence committed:
add scripts for lint and test

jazzsequence committed:
phpcbf fixes

jazzsequence committed:
linting fixes

jazzsequence committed:
add gh actions to run lint and tests

jazzsequence committed:
remove tests
these are testing things in the global library, not in this module

jazzsequence committed:
Merge branch 'main' into 3-phpunit-phpcs

RobLoach commented on phpunit.xml:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<phpunit bootstrap="tests/config/bootstrap.php" colors="true" cacheResult="false">
```
Could likely reference `vendor/autoload.php` directly rather than the small bootstrap.php file. Save having to have the tests/config folder.

jazzsequence replied:
Good call. It's not terribly likely we'll need to load anything else in the bootstrap.

jazzsequence committed:
update readme

jazzsequence committed:
remove the bootstrap file
if we aren't running any other bootstrap functions, we probably don't need this

jazzsequence committed:
require the vendor/autoload.php directly

jazzsequence committed:
don't forget the tests/ folder

jazzsequence committed:
add a global phpcs ruleset

jazzsequence committed:
update the composer commands and adds php syntax error checking

jspellman814 commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,22 @@
+name: smart_content_cdn
+on:
```
Suppose we could always change this later, but wondering, do we want to limit this Action to just one branch?

jazzsequence replied:
Probably not. That was Copilot being enthusiastic.

jazzsequence committed:
add build status badge to readme

jazzsequence committed:
run on any push

jazzsequence committed:
update github workflow

jazzsequence committed:
add name to checkout step

jazzsequence committed:
fix badge path
